[lldb][docs] Some fixes for lldbgdbremote.md. (#200079)
llvm/llvm-project: lldb/docs/resources/lldbgdbremote.md

Reviewer: So these actually need to be hex encoded. Which will look strange but is correct, because the response format follows:
> For a list of the key/value pairs in the response see the `qProcessInfoPID` packet documentation.

Reviewer: If we add an example where we send a triple, that would show the difference.

Reviewer: In fact: a note calling out this asymmetry would be good too.

"Note that triples sent in this packet are normal strings, but triples received in response are hex encoded strings. This difference was unintentional but cannot be changed at this point."

Author: > If we add an example where we send a triple, that would show the difference.

The third exchange in the example clarifies that triples in the sent packet are not hex encoded; the first two need changing.  I do not have a `x86_64-apple-macosx` system with me.   Is it okay if I replace the first two exchanges with

```
send packet: $qfProcessInfo:name_match:contains;name:656d616373;all_users:0;#21
read packet: $pid:4086;ppid:2681;uid:1000;gid:1000;euid:1000;egid:1000;name:2f7573722f62696e2f656d6163732d67746b;args:656d616373-2d2d6461656d6f6e;triple:7838365f36342d2d6c696e75782d676e75;#07
send packet: $qsProcessInfo#4f
read packet: $pid:146456;ppid:1;uid:1000;gid:1000;euid:1000;egid:1000;name:2f7573722f62696e2f656d6163732d67746b;args:2f7573722f62696e2f656d616373;triple:7838365f36342d2d6c696e75782d676e75;#e2
```
(which corresponds to
```
2 matching processes were found on "remote-linux"
 whose name contained "emacs"
PID    PARENT USER       TRIPLE                         NAME
====== ====== ========== ============================== ============================
4086   2681   mundkur    x86_64-unknown-linux-gnu       emacs-gtk
146456 1      mundkur    x86_64-unknown-linux-gnu       emacs-gtk
```
)

Author: Sorry, there is no existing third exchange, I was looking at something I added :-)

Author: Don't the first two send-read packet exchanges in the sample packet/response still need to be fixed?  The response packets are still showing triples as normal strings.  I asked if I could use the exchanges I captured from my `x86_64-unknown-linux-gnu` system above.

Reviewer: Sorry I missed that. Yes they should be hex and you can use your data (it's just an example after all).

Author: Done.

  be listed for all users, not just the user that the
  platform is running as
* `triple` - `string` -

>   An hex encoded triple string (hex encodings of `x86_64`, `x86_64-apple-macosx`, `armv7-apple-ios`)

Reviewer: I am confused because the code implies that it's a normal string.

In GDBRemoteCommunicationClient::FindProcesses:
```
      if (match_info.GetProcessInfo().GetArchitecture().IsValid()) {
        const ArchSpec &match_arch =
            match_info.GetProcessInfo().GetArchitecture();
        const llvm::Triple &triple = match_arch.GetTriple();
        packet.PutCString("triple:");
        packet.PutCString(triple.getTriple());
        packet.PutChar(';');
      }
```
The server also thinks it's a normal string:
```
      } else if (key == "triple") {
        match_info.GetProcessInfo().GetArchitecture() =
            HostInfo::GetAugmentedArchSpec(value);
      } else {
```
However, I do see it hex encoded in the logs when I get a **response**:
```
lldb             <  17> send packet: $qfProcessInfo#42
lldb             < 219> read packet: $pid:1427;ppid:1;uid:24900;gid:24900;euid:24900;egid:24900;name:2f75
73722f6c69622f73797374656d642f73797374656d64;args:2f6c69622f73797374656d642f73797374656d64-2d2d75736572;t
riple:616172636836342d2d6c696e75782d676e75;#ea
```

To be sending a triple to the server I think I need to use:
```
(lldb) platform process list --arch arm64
error: invalid combination of options for the given command
```
But it doesn't actually work.

Reviewer: It's because you have to give --arch and something else. Which is, a choice. Seems like it should work just fine on its own, but anyway:
```
(lldb) platform process list --arch arm64 --contains abc
lldb             <  93> send packet: $qfProcessInfo:name_match:contains;name:616263;all_users:0;triple:ar
m64-unknown-linux-gnu;#da
lldb             <   7> read packet: $E03#a8
error: no processes were found that contained "abc" on the "remote-linux" platform
```
It is not hex encoded.

So this packet is:
* Sending a normal triple string
* Expecting to receive a hex encoded triple string

Which isn't great but it is what it is so I think this part of the doc is actually correct.

Reviewer: https://github.com/llvm/llvm-project/issues/200138 for the command's issue.

```

Key value pairs are one of:
* `arch`: a string for the architecture, not needed if "triple" is specified
* `cputype`: is a number that is the mach-o CPU type that is being debugged (base 10)
* `cpusubtype`: is a number that is the mach-o CPU subtype type that is being debugged (base 10)

> * `triple`: a hex encoded string for the target triple (x86_64-apple-macosx) that can be used to specify arch + vendor + os in one entry

Reviewer: Add "for example": "target triple (for example `x86_64-apple-macosx`)"

```

Key value pairs are one of:
* `arch`: a string for the architecture, not needed if "triple" is specified
* `cputype`: is a number that is the mach-o CPU type that is being debugged (base 10)
* `cpusubtype`: is a number that is the mach-o CPU subtype type that is being debugged (base 10)

> * `triple`: a hex encoded string for the target triple (for example, hex encoding of `x86_64-apple-macosx`) that can be used to specify arch + vendor + os in one entry

Reviewer 2: ```suggestion
* `triple`: an ASCII hex encoded string for the target triple (for example, hex encoding of `x86_64-apple-macosx`) that can be used to specify arch + vendor + os in one entry
```

* `euid` - `integer` - A string value containing the decimal effective user ID
* `egid` - `integer` - A string value containing the decimal effective group ID
* `name` - `ascii-hex` - An ASCII hex string that contains the name of the process

> * `triple` - `string` - A hex encoded target triple (hex encodings of `x86_64-apple-macosx`, `armv7-apple-ios`)

Reviewer: Again, "for example ...". Don't want to imply that those 2 triples are the only ones you can expect.

* `euid` - `integer` - A string value containing the decimal effective user ID
* `egid` - `integer` - A string value containing the decimal effective group ID
* `name` - `ascii-hex` - An ASCII hex string that contains the name of the process

> * `triple` - `string` - A hex encoded target triple (for example, hex encodings of `x86_64-apple-macosx`, `armv7-apple-ios`)

Reviewer 2: We generally call this "ASCII hex". Let's mirror the line above:
```suggestion
* `triple` - `ascii-hex` - An ASCII hex string that contains the target triple (for example, `x86_64-apple-macosx`, `armv7-apple-ios`)
```
